larrecodnn/ImagePatternAlgs/NuSonic/Triton/triton_utils.cc

```diff
@@ -0,0 +1,37 @@
+#include "larrecodnn/ImagePatternAlgs/NuSonic/Triton/triton_utils.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+#include "canvas/Utilities/Exception.h"
+
+#include <sstream>
+#include <experimental/iterator>
+
+namespace triton_utils {
+
+  template <typename C>
+  std::string printColl(const C& coll, const std::string& delim) {
+    if (coll.empty())
+      return "";
+    std::stringstream msg;
+    //avoid trailing delim
+    std::copy(std::begin(coll), std::end(coll), std::experimental::make_ostream_joiner(msg, delim));
```

Comment:
The usage here could be adjusted a little:

```c++
using std::begin;
using std::end;
std::copy(begin(coll), end(coll), std::experimental::make_ostream_joiner(msg, delim));
```

That way someone's custom overload of `begin` or `end` can be used instead of mandating the C++ STL ones.

Reply:
done